bzt/utils.py

```diff
@@ -1143,10 +1250,15 @@ def mirrors(self):
         self.log.debug("Retrieving mirrors from page: %s", self.base_link)
         downloader = ExceptionalDownloader()
         try:
-            tmp_file = downloader.get(self.base_link)[0]
-            with open(tmp_file) as fds:
-                self.page_source = fds.read()
+            if self.http_client:
+                response = self.http_client.request('GET', self.base_link)
```

Comment:
Why not pass self.http_client to downloader that is ExceptionalDownloader?

Reply:
It's because downloader does some extra steps (allocate a tempfile, store the response in it), that aren't necessary for just retrieving HTML of page with links to mirrors. `HTTPClient` allows writing it in two lines.

In the future, once every `RequiredTool` and `MirrorManager` uses `HTTPClient` inside, this `if` will be removed. Until then, it has to be there for compatibility.